packages/block-library/src/paragraph/edit.js

```diff
@@ -64,20 +65,54 @@ function ParagraphBlock( {
 		style: { direction },
 	} );
 
+	const updateContent = useCallback(
+		( newContent ) => setAttributes( { content: newContent } ),
+		[ setAttributes ]
+	);
+
+	const splitParagraph = useCallback(
+		( value, isOriginal ) => {
+			let newAttributes;
+
+			if ( isOriginal || value ) {
+				newAttributes = {
+					...attributes,
+					content: value,
+				};
+			}
+
+			const block = createBlock( name, newAttributes );
+
+			if ( isOriginal ) {
+				block.clientId = clientId;
+			}
+
+			return block;
+		},
+		[ attributes, createBlock ]
+	);
+
+	const updateAlign = useCallback( ( newAlign ) =>
+		setAttributes( { align: newAlign }, [ setAttributes ] )
+	);
+
+	const setDirection = useCallback(
+		( newDirection ) => setAttributes( { direction: newDirection } ),
+		[]
```

Comment:
`setAttribute` should be in the dep array.

Reply:
[The docs](https://reactjs.org/docs/hooks-reference.html#usecallback) don't touch on adding methods as dependencies on the `useCallback` hook — in fact, the example doesn't. Is `setAttributes`'s identity changing at some point to make this necessary?

Reply:
We don't have a consensus of this yet, but I strongly recommend we adding it. See https://github.com/WordPress/gutenberg/pull/32469#discussion_r647627917

It's also for consistency that other `useCallback`s are including it but this doesn't. Also, this could potentially fix bugs if we accidentally pass a different function with the same name here (like a wrapper or something).

Reply:
Good catches, anything that is used in the callback should be marked as a dependency. There's an exception for React's useState hook, eg 

> The identity of the dispatch function from useReducer is always stable

https://reactjs.org/docs/hooks-faq.html#what-can-i-do-if-my-effect-dependencies-change-too-often

and when setting state via useState there's the option to use a functional update to avoid a closure with a stale value: https://reactjs.org/docs/hooks-reference.html#functional-updates

Reply:
Going to mark this as resolved, since I ended up dropping these changes